### ISSUE TITLE
Issues Communicating with a Ektapro 5000 projector

### ISSUE THREAD

**Issue by thesoundofthesound** [bullpup/ektaprogui#1]
Hello, 

I've code this code and am really interested in trying to use it. 

I'm running the code on a Mac and attempting to send commands to a Ektapro 5000 projector (which has a single P-Bus in port). I've been attempting to update the code, which at this point is quite old. Some minor tweaking has gotten it to run in principle, with the GUI showing up. 

However, I am running into some problems. Mainly, there's no communication occurring. I've been able to send a command out to the projector, but nothing is received (i.e. no device information is received when running initDevices() found on line 60.

It's a long shot, but any help / guidance would be really appreciated!

**Comment by julian-hoch** [comment]
Unfortunately I neither have access to a Mac nor an Ektapro projector right now, but I would first make sure that your PC actually can access the projector via the serial port. I would do this with a tool that can directly probe the device, to make sure its not simply a driver or connectivity issue.
Good luck!

**Comment by thesoundofthesound** [comment]
Thanks Julian. No problem. That's a good idea to check connectivity. 

I've used the program CoolTerm to perform a loopback test (wiring pins 2 and 3 together). I can confirm that it's a cable/adapter issue that I'm currently facing. Tx and Rx aren't coming out of the correct pins for the projector. 

I'm using a Null Modem cable, which is good what it's supposed to do. But I had to use a gender changer adapter to plug into the Ektapro. Funnily enough, that's where the issue is: the gender changer somehow has change the Null Modem pins to a 'standard' serial pin out... 

I have other cables and adapters on order. Hopefully this will resolve part of the issue and I can get your code working!
